### PR TITLE
Prevent duplicate Hotseat player names

### PR DESCRIPTION
**Pull request by strathmeyer**
Fixes tobymao/18xx#71

After I prevented strict duplicates, I realized it was still possible to add names that would appear identical such as `<space>Player<space>1` and `Player<space><space>1`. So I added a second commit to trim whitespace out of the name. That took the longest because I couldn't figure out why `.gsub(/[[:space:]]+/, " ")` wasn't working. Turns out it's not supported by Opel! 😢 

I haven't added any tests yet. I couldn't find anything existing in the `/spec` folder that mentioned the Hotseat feature or the CreateGame class. Let me know if that's a merge blocker.

### CODE REVIEW

File: assets/app/view/create_game.rb [hunk 1]
```diff
@@ -110,7 +114,7 @@ def submit
         end
 
         create_hotseat(
-          players: players,
+          players: players.map { |name| { name: name } },
```
Comment (tobymao) [inline assets/app/view/create_game.rb:117]: why is this changed? was it always broken?

File: assets/app/view/create_game.rb [hunk 2]
```diff
@@ -95,7 +95,11 @@ def submit
         players = params
           .select { |k, _| k.start_with?('player_') }
           .values
-          .map { |name| { name: name } }
+          .map { |name| name.gsub(/\s+/, ' ').strip }
```
Comment (tobymao) [inline assets/app/view/create_game.rb:98]: ehhh i don't think it's necessary to do this since the names are user input

File: assets/app/view/create_game.rb [hunk 3]
```diff
@@ -95,7 +95,11 @@ def submit
...
+
+        if players.any? { |name| players.count(name) > 1 }
```
Comment (tobymao) [inline assets/app/view/create_game.rb:100]: i think this could be simplified as if players.uniq.size < players.size

Comment (tobymao) [reply assets/app/view/create_game.rb:100]: return store(:flash_opts) if player.uniq.size < players.size

File: assets/app/view/create_game.rb (hunk 1 above)
Comment (strathmeyer) [reply assets/app/view/create_game.rb:117]: Nope, but when I was doing the whitespace stripping below, I ended up with code like `.map { |name| name[:name].gsub(/\s+/, ' ').strip }` which was just too many "names" in a row.

I thought, "why not map to parameter format down inside `create_hotseat` where it's required, and let the rest of the code deal with the player names as an array of strings." Perhaps I should have renamed the array of strings to `player_names`?

File: assets/app/view/create_game.rb (hunk 2 above)
Comment (strathmeyer) [reply assets/app/view/create_game.rb:98]: If our stance is that users should avoid Hotseat usernames that result in indistiguishable display strings, we should just close #71 as won't fix.

File: assets/app/view/create_game.rb (hunk 3 above)
Comment (strathmeyer) [reply assets/app/view/create_game.rb:100]: Sounds like you have a one line fix for #71 in mind. I think it'll save us both time if you commit that directly to master, rather than me updating this PR.

File: assets/app/view/create_game.rb (hunk 2 above)
Comment (tobymao) [reply assets/app/view/create_game.rb:98]: well the difference is that white space is only confusing whereas duplicate names breaks the game

File: assets/app/view/create_game.rb (hunk 1 above)
Comment (tobymao) [reply assets/app/view/create_game.rb:117]: ah i see it now thanks, this is fine

File: assets/app/view/create_game.rb (hunk 3 above)
Comment (tobymao) [reply assets/app/view/create_game.rb:100]: i will just merge this and update it

File: assets/app/view/create_game.rb (hunk 2 above)
Comment (strathmeyer) [reply assets/app/view/create_game.rb:98]: You're right. I was thinking that having two indistinguishable players also effectively breaks the game for a Hotseat play.

I usually try to avoid solving a bigger problem than the one described in the issue. I'll be extra careful about that in future PRs. 👍

File: assets/app/view/create_game.rb (hunk 3 above)
Comment (strathmeyer) [reply assets/app/view/create_game.rb:100]: Thank you Toby! 🚂

Comment (tobymao) [reply assets/app/view/create_game.rb:100]: thanks for your first PR, looking forward to your log fixes
